src/modules/test/README.md

```diff
@@ -1,60 +1,193 @@
-
-Every module that includes a [test recipe](https://github.com/Azure/azure-osconfig/tree/main/src/modules/test/recipes) and [MIM](https://github.com/Azure/azure-osconfig/tree/main/src/modules/mim) is automatically included in the test configuration (testplate.json).
-# Prerequisites
-Although `modulestest` does not require any runtime dependencies, generating the Test Recipe configuration (testplate.json) does require the [jq](https://github.com/stedolan/jq) application to be installed when building OSConfig.
-
-# Test Naming Composition
-Authored test recipes use the following test naming composition when registering with the Google Test framework. This is useful diagnostics as the name is used in test reports. `ModulesTest` is the test suite name and is further composed with the `ModuleName` (defined by the directory/file name), `ComponentName` and `ObjectName` (`<null>` is used if either are empty strings).
-
-## Example - Test Naming Convention
-For example, the following test recipe (sample Tpm recipe) will create the following test names:
- * `ModulesTest.tpm.Tpm.tpmStatus`
- * `ModulesTest.tpm.<null>.tpmStatus`
- * `ModulesTest.tpm.<null>.<null>`
-```
-[
-  {
-    "ComponentName": "Tpm",
-    "ObjectName": "tpmStatus",
-    "Desired": false,
-    "ExpectedResult": 0,
-    "WaitSeconds": 0
-  },
-  {
-    "ComponentName": "",
-    "ObjectName": "tpmStatus",
-    "Desired": false,
-    "ExpectedResult": 22,
-    "WaitSeconds": 0
-  },
-  {
-    "ComponentName": "",
-    "ObjectName": "",
-    "Desired": false,
-    "ExpectedResult": 22,
-    "WaitSeconds": 0
+# `moduletest`
+
+`moduletest` is a functional testing tool that allows developers to easily validate the functionality of a module using a test definition to specifiy a series of steps. These steps con be used to validate the input/output of a module and validate that the returned payloads and status codes are as expected.
+
+## Usage
+
+The `moduletest` command line tool is used to *execute* a test definition.
+
+> By default, modules are loaded from `/usr/lib/osconfig`. Modules can be loaded from a different directory using the `--bin` flag.
+
+> For convenience, module binaries and test definitions are collected under the `build/` directory when building `azure-osconfig`:
+> - Shared-object binaries are placed under `build/modules/bin`
+> - Test definitions are placed under and renamed to match their module respective shared-object binary `build/test/<module>.json`
+
+```bash
+$ moduletest <file>... [--bin <path>] [--verbose] [--help]
+
+# Example
+$ moduletest path/to/module/test.json
+$ moduletest path/to/module/test.json --bin /usr/lib/osconfig
+$ moduletest path/to/module/test1.json path/to/module/test2.json

```

Comment:
We discussed to not do this anymore. To have as the only possible command line arguments to modulestest the test recipes.

Reply:
Change "test.json" to "ModuleTestRecipe.json" or similar?